### PR TITLE
BGE uses dynamic fields for form and grid

### PR DESCRIPTION
**Pull request by bethbrains**
# Critical Changes

# Changes
- BGE Form and Grid fields both draw dynamically from Active_Fields__c on Batch record

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by bethbrains** [comment]
**lurch: attach W-028587

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007fNyUAAU" target="_blank">W-028587</a>

File: src/classes/BGE_DataImportBatchEntry_CTRL.cls [hunk 1]
```diff
@@ -165,6 +170,46 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
...
     }
 
+    /*******************************************************************************************************
+    * @description Retrieves the related list of columns defined by users in the Batch record Active_Fields__c
+    * @return list of Columns to use in BGE_DataImportBatchEntry table
+    */
+    private static List<Column> getUserDefinedColumns(Id batchId) {
+        List<Column> columns = new List<Column>();
+
+        DataImportBatch__c batch = [SELECT Active_Fields__c FROM DataImportBatch__c WHERE Id = :batchId];
```
[the remark this answers is not in the document]
Comment (randi274) [reply src/classes/BGE_DataImportBatchEntry_CTRL.cls]: @lmeerkatz we sure could do that! Is it worth the new method for a 2 line change here?

**Comment by cpose** [comment]
QA Approved, Merging.

**Pull request merged**